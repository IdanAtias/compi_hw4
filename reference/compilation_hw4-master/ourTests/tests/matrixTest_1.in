matrix m1(2,3) = [1,2,3;4,5,6];
matrix m2(2,3) = m1, m3(3,2),m5(3,2)=[1,2;3,4;5,6], m4(2,2) = [1,-2;-3,4];
matrix m6(2,1) = [-1;2];

m1::m4::m4;
m1*m3;
m5*m4*m4;
m1+m2;
m1-m2;
transpose(m1);
m4^(2);
m4^(3);

(m6::m6*transpose(m5))*(m5*m4^(2));
((m4^(2))*m1-(m1+m1))*transpose(m1);

([-1;2]::m6*transpose([1,2;3,4;5,6]))*(m5*m4^(2));
(([1,-2;-3,4]^(2))*m1-(m1+m1))*transpose([1,2,3;4,5,6]);

[1,2,3;4,5,6]::m4::[1,-2;-3,4];
[1,2,3;4,5,6]*m3;
m5*m4*[1,-2;-3,4];
m1+[1,2,3;4,5,6];
[1,2,3;4,5,6]-m2;
transpose([1,2,3;4,5,6]);
[1,-2;-3,4]^(2);
[1,-2;-3,4]^(3);
