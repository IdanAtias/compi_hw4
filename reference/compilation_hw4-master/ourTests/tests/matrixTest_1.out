m1=
1,2,3
4,5,6
m2=
1,2,3
4,5,6
m3=
0,0
0,0
0,0
m5=
1,2
3,4
5,6
m4=
1,-2
-3,4
m6=
-1
2
1,2,3,1,-2,1,-2
4,5,6,-3,4,-3,4
0,0
0,0
-23,34
-39,58
-55,82
2,4,6
8,10,12
0,0,0
0,0,0
1,4
2,5
3,6
7,-10
-15,22
37,-54
-81,118
947,-1410
-1894,2820
-250,-610
430,1060
947,-1410
-1894,2820
-250,-610
430,1060
1,2,3,1,-2,1,-2
4,5,6,-3,4,-3,4
0,0
0,0
-23,34
-39,58
-55,82
2,4,6
8,10,12
0,0,0
0,0,0
1,4
2,5
3,6
7,-10
-15,22
37,-54
-81,118
